### ISSUE TITLE
silx view track_order

### ISSUE THREAD

**Issue by mretegan** [silx-kit/silx#3128]
Any ideas why silx view doesn't keep the order of the keys?

```python
import h5py

h5py.get_config().track_order = True

filename = "test.h5"
with h5py.File(filename, "w") as h5:
    params = h5.create_group("params")
    params["z"] = 10
    params["a"] = 2
    print(params.keys())

with h5py.File(filename, "r") as h5:
    print(h5["params"].keys())
``` 

```
<KeysViewHDF5 ['z', 'a']>
<KeysViewHDF5 ['z', 'a']>
```

`silx view test.h5`

<img width="1199" alt="Screenshot 2020-06-15 at 14 46 45" src="https://user-images.githubusercontent.com/93124/84659010-0c147f80-af17-11ea-8a9f-738dab94038a.png">

**Comment by vasole** [comment]
It is quite normal for a "File browser" like application to use alphabetical order irrespective of when the "Files" have been created.

I would not expect silx view to respect that. Reading the file is another business.

**Comment by mretegan** [comment]
But if a certain order was imposed on creation, why not use it? 

When the file is created, you can choose if you want to keep or not the order. If you don't, then you will have the alphabetical sorting anyways.

Could this be an option for the viewer?

**Comment by vasole** [comment]
It would be very confusing for a user to hunt for a set of names in arbitrary order.

**Comment by mretegan** [comment]
I agree that ordering keys alphabetically makes sense in most situations. However, I think that if someone decided to set `retain_order=True` to have a specific order, the viewer should at least have an option to respect that. For example, Argos (https://github.com/titusjan/argos) respects the order by default. It is true that other tools like `h5dump` don't.

**Comment by mretegan** [comment]
Anyway, if the choice was made for silx view, I can live with that. I was just surprised that it did not behave the way that I expected. I was actually trying to create .hdf5 files that have a certain order and I did not understand why I was having them ordered alphabetically in silx view.

**Comment by vallsv** [comment]
The sorting is done by the model used by the treeview. So you could disable it if you like. Or this could be exposed on top of the treeview as an action. I don't think it is a complex thing.

I think it is not possible to check if an hdf5 group was created to store or not the order. I think there was kind of weird things done by h5py, i can be wrong.

**Comment by t20100** [comment]
I'm having a look at it, it is simple to remove the sorting:
use `treeModel` instead of `treeModel2` here https://github.com/silx-kit/silx/blob/master/silx/app/view/Viewer.py#L95

Providing an action on top of the tree toggle sorting sounds good to me.

**Comment by t20100** [comment]
PR #3132 proposes a solution to that. Feedback (and icon!) welcome.

**Comment by t20100** [comment]
Closed by PR #3132 on branch 0.13
